ops/createComponent.js

```diff
@@ -8,33 +8,36 @@ import camelToSnake from "../lib/camelToSnake.js"
 import importsBuilder from "./importsBuilder.js"
 
 export default async function createComponent({
-  path,
+  path = ``,
   name,
   body,
   imports,
   args,
   css,
 }) {
-  const realPath = path ? `app/${path}` : `app`
-  path = path || ``
-
-  const file = `${realPath}/components/${name}.js`
-  const demo = `${realPath}/components/${name}/demo.js`
-  const cssFile = `${realPath}/components/${name}/css.js`
+  const isInApp = process.cwd().indexOf(`/app`) !== -1
+  const isAppPath = path?.startsWith(`app`)
+  const resolvePath = (isInApp || isAppPath) ? path : `app/` + path
```

Comment:
@mikabytes maybe i need you to jump-in here, it's not about solution, but i cannot make a decision here.
The Solution:
- I just need move those 3 lines of code to `cli` level, which mean everything related to path resolved, path transform will be handled in `cli` => create command level will receive whatever path cli pass into, execute the rest of logic (create component/css/imports/...)

The Issue:
- All of test case of `ops` and `cli` need to be rewrite, because:
- `cli/parse.test.js` expect object change (we already resolve the `path` with prefix `app`)
- `ops/*.text.js` now test case must give resolved path to create methods, create method itself doesn't resolve path anymore (as solution given)

Example:
- Given: 
   + cwd: "./"
   + command: "tiden create component myComponent one"
- Expected:
   + `cli` check that namespace doesn't point to app path, cwd not in app dir also => implicit resolve namespace to `app/one`
   + `ops` receive namespace as `app/one`, know nothing about what `cli` did
- Issue: all test case in `ops/createComponents.test.js` fail, due to calling direct to createComponent methods instead of go through cli

Reply:
This issue is becoming more complicated for every day, haha :) But it's good, your investigations are leading us forward to better arch.

I will need to think deeply about this. But a few initial impressions:

1. `cli/parse.js` should only ever parse. Nothing else. So we cannot make the "translation" of namespace directory here.
2. Anything `ops` should receive the full details needed to do their work. They should not do modifications based on `cwd`.

So it seems there might be a need for a 3rd layer somewhere. I'll spend some time trying to figure this out now.

Reply:
> I will need to think deeply about this. But a few initial impressions:
> 
> 1. cli/parse.js should only ever parse. Nothing else. So we cannot make the "translation" of namespace directory here.
> 2. Anything ops should receive the full details needed to do their work. They should not do modifications based on cwd.
> 

couldn't agree more, right here waiting for you my man

Reply:
Moving this conversation to https://github.com/tidenjs/tiden/issues/19